components/src/core/user-menu/user-menu.tsx

```diff
@@ -0,0 +1,103 @@
+        avatar?: ViewStyle;
+    };
+};
+
+const useStyles = (
+    theme: ReactNativePaper.Theme
+): StyleSheet.NamedStyles<{
+    root: ViewStyle;
+    avatar: ViewStyle;
+}> =>
+    StyleSheet.create({
+        root: {
+            backgroundColor: theme.colors.surface,
+        },
+        avatar: {
+            width: 40,
+            height: 40,
+            borderRadius: 40,
+        },
+    });
+
+export const UserMenu: React.FC<UserMenuProps> = (props) => {
+    const theme = useTheme();
+    const { avatar, backgroundColor, fontColor, iconColor, menuTitle, menuSubtitle, menuItems, styles = {} } = props;
+    const [showBottomSheet, setShowBottomSheet] = useState(false);
+    const defaultStyles = useStyles(theme);
+
+    const openMenu = (): void => {
+        if (menuItems) setShowBottomSheet(true);
+    };
+
+    const closeMenu = (): void => {
+        setShowBottomSheet(false);
+    };
+
+    const getMenu = useCallback(
+        (): JSX.Element => (
+            <>
+                {menuTitle && (
+                    <>
+                        <InfoListItem
+                            hidePadding
+                            title={menuTitle || ''}
+                            subtitle={menuSubtitle}
+                            leftComponent={
+                                <View style={[defaultStyles.avatar, styles.avatar, { marginLeft: 16 }]}>{avatar}</View>
+                            }
+                            fontColor={fontColor}
+                            backgroundColor={backgroundColor}
+                        />
+                        <Divider />
+                    </>
+                )}
+                {menuItems &&
+                    menuItems.map((menuItem: InfoListItemProps, index: number) => (
+                        <InfoListItem
+                            {...menuItem}
+                            key={index}
+                            onPress={(): void => {
+                                closeMenu();
+                                if (menuItem.onPress) menuItem.onPress();
+                            }}
+                            iconColor={iconColor || menuItem.iconColor}
+                            fontColor={fontColor || menuItem.fontColor}
+                            backgroundColor={backgroundColor || menuItem.backgroundColor}
+                        />
+                    ))}
+            </>
+        ),
+        [menuItems, menuTitle, menuSubtitle, iconColor, fontColor, backgroundColor]
+    );
+
+    return (
+        <View style={[defaultStyles.root, styles.root]}>
+            <TouchableWithoutFeedback onPress={(): void => openMenu()} testID={'avatar'}>
+                <View style={[defaultStyles.avatar, styles.avatar]}>{avatar}</View>
+            </TouchableWithoutFeedback>
+            <BottomSheet show={showBottomSheet} backgroundColor={backgroundColor} onClose={(): void => closeMenu()}>
+                {getMenu()}
+            </BottomSheet>
+        </View>
```

Comment:
Wondering if all of these wrapper elements are necessary. Could we reduce this to something like:
```suggestion
        <>
            <TouchableWithoutFeedback onPress={(): void => openMenu()} testID={'avatar'} styles={[defaultStyles.root, styles.root]}>
                {avatar}
            </TouchableWithoutFeedback>
            <BottomSheet show={showBottomSheet} backgroundColor={backgroundColor} onClose={(): void => closeMenu()} styles={{root: styles.bottomsheet}}>
                {getMenu()}
            </BottomSheet>
        </>
```